### PR TITLE
Split `simple-smoke-test.sh` to be used elsewhere [DI-215]

### PR DESCRIPTION
**Pull request by JackPGreen**
`simple-smoke-test.sh` is a useful suite of tests to validate a packaged Hazelcast instance - but is Docker-centric.

- Refactored to split the script into two halves:
   - `abstract-simple-smoke-test` that tests against an abstract Hazelcast instance
  - `simple-smoke-test` a Docker-specific implementation
- Refactored the metadata checks to search the instance's logs for the variant and version (i.e. OSS/EE, `5.5.0` etc) rather than inspecting the containers' filesystem
   - an existing test elsewhere in the codebase already does this so we already have some dependency on keeping the output consistent, although I can't currently find the test in question
- Added extra coverage from [`hazelcast-packaging`'s `check-hazelcast-health.sh`](https://github.com/hazelcast/hazelcast-packaging/blob/bd417d0604b57ae2f72234d8781929fdf86b0687/check-hazelcast-health.sh)

Required for [DI-215](https://hazelcast.atlassian.net/browse/DI-215)

[DI-215]: https://hazelcast.atlassian.net/browse/DI-215?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ldziedziul — APPROVED** [review]
Nice work! Although in the long term we should think about extracting these check to a seperate repository that could be used in docker, packaging workflows but also for pre/post-release verification, just a thought

**Comment by JackPGreen** [comment]
@nishaatr are you able to review, please?

File: .github/scripts/simple-smoke-test.sh [hunk 1]
```diff
@@ -28,81 +15,39 @@ function remove_container_if_exists() {
...
-      echoerr "${image} version was ${version}, not ${expected_version} as expected"
-      exit 1
-    fi
-}
-
-function test_map_read_write() {
-    local image=$1
-    local expected_distribution_type=$2
-
+function start() {
```
Comment (nishaatr) [inline .github/scripts/simple-smoke-test.sh:18]: better `start_container()`

File: .github/scripts/simple-smoke-test.sh [hunk 2]
```diff
@@ -28,81 +15,39 @@ function remove_container_if_exists() {
...
-  export PATH=${PATH}:${HOME}/.hazelcast/bin
-  clc config add default cluster.name=dev cluster.address=localhost
+function get_hz_logs() {
+    docker logs "${container_name}"
 }
 
-# Prints the given message to stderr
-function echoerr() {
-  echo "ERROR - $*" 1>&2;
+function stop() {
```
Comment (nishaatr) [inline .github/scripts/simple-smoke-test.sh:27]: better `stop_container()`

File: .github/scripts/simple-smoke-test.sh [hunk 3]
```diff
@@ -28,81 +15,39 @@ function remove_container_if_exists() {
...
 
-
-install_clc
-test_docker_image "$@"
+image=$1
+container_name=$2
+input_distribution_type=$3
+expected_version=$4
+
+case "${input_distribution_type}" in
```
Comment (nishaatr) [inline .github/scripts/simple-smoke-test.sh:37]: nit - can we move this to a function()?
Looks little naked

**Review by nishaatr — CHANGES_REQUESTED** [review]

File: .github/scripts/abstract-simple-smoke-test.sh [hunk 4]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o pipefail
+
```
[the remark this answers is not in the document]
Comment (JackPGreen) [reply .github/scripts/abstract-simple-smoke-test.sh:4]: 9e520430a59c86afa2179eb2b41dbbc1318271d4

File: .github/scripts/abstract-simple-smoke-test.sh [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+# Performs simple validation tests on an already-running Hazelcast instance
+# Abstract as could be from Docker, Homebrew, local binary etc
+function test_package() {
+    local expected_distribution_type=$1
+    local expected_version=$2
+
+    test_health
+    test_map_read_write
+
+    # Deliberately last as it doesn't block-and-wait until the instance is initialized
```
[the remark this answers is not in the document]
Comment (JackPGreen) [reply .github/scripts/abstract-simple-smoke-test.sh]: 6240dd25988dc7c4360eddabcbb63b611ffb5628

File: .github/scripts/simple-smoke-test.sh (hunk 1 above)
Comment (JackPGreen) [reply .github/scripts/simple-smoke-test.sh]: 41e631bf6f6a442f6542b5c234540e82eb44513c

File: .github/scripts/simple-smoke-test.sh (hunk 2 above)
Comment (JackPGreen) [reply .github/scripts/simple-smoke-test.sh]: 41e631bf6f6a442f6542b5c234540e82eb44513c

File: .github/scripts/simple-smoke-test.sh (hunk 3 above)
Comment (JackPGreen) [reply .github/scripts/simple-smoke-test.sh]: e676ec1f3a41b47de0fcd9a4f8b9503d1fa911bf

**Review by nishaatr — APPROVED** [review]

**Pull request merged**